public.tf

```diff
@@ -0,0 +1,74 @@
+module "public_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
+  namespace  = "${var.namespace}"
+  name       = "${var.availability_zone}"
+  attributes = ["public"]
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_subnet" "public" {
+  count             = "${var.type == "public" ? length(var.names) : 0}"
+  vpc_id            = "${var.vpc_id}"
+  availability_zone = "${var.availability_zone}"
+  cidr_block        = "${cidrsubnet(var.cidr_block, ceil(log(length(var.names), 2)), count.index)}"
```

Comment:
To future proof this, I wonder if we should add a `max_subnets` parameter instead and always base this off of the max. But we only ever create `length(var.names)` subnets.  That way we can add subnets without causing renumbering.

Reply:
Added.